src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Color;
+
+  sideA: number;
+
+  sideB: number;
+
+  sideC: number;
+
+  constructor(color: Color, sideA: number, sideB: number, sideC: number) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('any length is <= 0');
+    }
+
+    if (
+      sideA >= sideB + sideC ||
+      sideB >= sideA + sideC ||
+      sideC >= sideA + sideB
+    ) {
+      throw new Error(
+        'the longest side of a triangle is >= than a sum of two others',
+      );
```

Comment:
The error message should be more descriptive. Consider providing details about what went wrong. For example: 'The sum of the lengths of any two sides must be greater than the length of the remaining side.'